### PR TITLE
Feat 88 zip vids

### LINKED ISSUE #88: Encrypt video folder before exporting to AWS
# User story
As a user, I want to encrypt the video data folder, so I can manage who can access it.

# Acceptance criteria
- [x] When the ephys job pipeline is run, the job that uploads the video folder encrypts the video folder first.

# Sprint Ready Checklist 
1. - [x] Acceptance criteria defined 
2. - [x] Team understands acceptance criteria 
3. - [x] Team has defined solution / steps to satisfy acceptance criteria 
4. - [x] Acceptance criteria is verifiable / testable 
5. - [x] External / 3rd Party dependencies identified
6. - [x] Ticket is prioritized and sized

# Notes
We can try using pyminizip, but it might require extra installation in a windows machine.

### PR DESCRIPTION
**Pull request by jtyoung84**
Closes #88

Limitations:
- People can still see file names.
- If secret_name is not defined in config file, then the videos are not encrypted (I can change this if this isn't the desired behavior, i.e., I can make it fail if no encryption password is given.)

### CODE REVIEW

File: src/aind_data_transfer/writers.py [hunk 1]
```diff
@@ -95,8 +115,22 @@ def copy_and_clip_data(src_dir, dst_dir, stream_gen, n_frames=100):
         videos_path = dst_dir / "Videos"
         videos_path_l = dst_dir / "videos"
         if os.path.isdir(videos_path):
-            new_videos_path = dst_dir / ".." / "videos"
-            shutil.move(videos_path, new_videos_path)
+            new_videos_path = dst_dir / ".." / "videos.zip"
```
Comment (dyf) [inline src/aind_data_transfer/writers.py:118]: I think we should compress the videos individually and leave the file names the way they were. It's fine to be able to see file names here.

**Review by dyf — CHANGES_REQUESTED** [review]
See comment.

**Comment by dyf** [comment]
@jtyoung84 One more thing.  Can you try opening one of these videos in CodeOcean? I was able to video an unencrypted AVI using `from IPython.display import Video`, but would like to see what the workflow is.

**Comment by jtyoung84** [comment]
> @jtyoung84 One more thing. Can you try opening one of these videos in CodeOcean? I was able to video an unencrypted AVI using `from IPython.display import Video`, but would like to see that this works after you've zipped.

It's working locally by using:

```
import pyminizip
from aind_data_transfer.util.s3_utils import get_secret
import json
from IPython.display import Video

vid_pass = json.loads(get_secret(***** , "us-west-2"))["password"]
directory_to_extract_video_to = None # None for current working dir
pyminizip.uncompress("PATH_TO_VIDEO/some_vid.avi.zip", 
                     vid_pass, 
                     None, 
                     1)
Video("some_vid.avi", embed=True)
```

File: src/aind_data_transfer/writers.py (hunk 1 above)
Comment (jtyoung84) [reply src/aind_data_transfer/writers.py]: Done

**Comment by dyf** [comment]
> > @jtyoung84 One more thing. Can you try opening one of these videos in CodeOcean? I was able to video an unencrypted AVI using `from IPython.display import Video`, but would like to see that this works after you've zipped.
> 
> It's working locally by using:
> 
> ```
> import pyminizip
> from aind_data_transfer.util.s3_utils import get_secret
> import json
> from IPython.display import Video
> 
> vid_pass = json.loads(get_secret(***** , "us-west-2"))["password"]
> directory_to_extract_video_to = None # None for current working dir
> pyminizip.uncompress("PATH_TO_VIDEO/some_vid.avi.zip", 
>                      vid_pass, 
>                      None, 
>                      1)
> Video("some_vid.avi", embed=True)
> ```

How long does it take to run?

**Comment by jtyoung84** [comment]
> > > @jtyoung84 One more thing. Can you try opening one of these videos in CodeOcean? I was able to video an unencrypted AVI using `from IPython.display import Video`, but would like to see that this works after you've zipped.
> > 
> > 
> > It's working locally by using:
> > ```
> > import pyminizip
> > from aind_data_transfer.util.s3_utils import get_secret
> > import json
> > from IPython.display import Video
> > 
> > vid_pass = json.loads(get_secret(***** , "us-west-2"))["password"]
> > directory_to_extract_video_to = None # None for current working dir
> > pyminizip.uncompress("PATH_TO_VIDEO/some_vid.avi.zip", 
> >                      vid_pass, 
> >                      None, 
> >                      1)
> > Video("some_vid.avi", embed=True)
> > ```
> 
> How long does it take to run (relative to uncompressed)?

I tested it with a relatively small video clip of around 20MB. `pyminizip` uncompresses the zipped file into a folder. I used a compression level of 5 (out of 1 to 9), so the speed can be improved at the cost of compression

**Review by dyf — APPROVED** [review]
ship it

**Pull request merged**